FrostyEditor/Windows/MainWindow.xaml.cs

```diff
@@ -386,19 +386,25 @@ private void launchButton_Click(object sender, RoutedEventArgs e)
             // setup ability to cancel the process
             CancellationTokenSource cancelToken = new CancellationTokenSource();
 
-            Random r = new Random();
-            string editorModName = $"EditorMod{r.Next(1000, 9999).ToString("D4")}.fbmod";
             launchButton.IsEnabled = false;
 
             // get all mods
             List<string> modPaths = new List<string>();
-
+            // Remove existing mods, because unhandled exceptions will cause mods to remain
             DirectoryInfo modDirectory = new DirectoryInfo($"Mods/{ProfilesLibrary.ProfileName}");
             foreach (string modPath in Directory.EnumerateFiles($"Mods/{ProfilesLibrary.ProfileName}/", "*.fbmod", SearchOption.AllDirectories))
-                modPaths.Add(Path.GetFileName(modPath));
+                File.Delete(modPath);
```

Comment:
this would get rid of all the mods in the folders, not just the editor mods

Reply:
there still needs to be modPaths.Add() for any other mods

Reply:
> there still needs to be modPaths.Add() for any other mods

It's done and I've successfully tested it in GW2